strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/authorization/AuthorizationConfigFileManager.java

```diff
@@ -45,4 +55,58 @@ public ConfigurationResourceResolver getConfigurationResourceResolver()
         return configurationResourceResolver;
     }
 
+    @Override
+    public synchronized void store(AuthorizationConfigDto configuration) throws IOException
+    {
+        Map<SystemRole, String> changed = getChangedRestrictedRoles(configuration);
+        if (changed.size() > 0)
+        {
+            String message = String.join("; ", changed.values());
+            throw new RuntimeException(message);
+        }
+        super.store(configuration);
+    }
+
+    public Map<SystemRole, String> getChangedRestrictedRoles(AuthorizationConfigDto configuration) throws IOException {
+        Map<SystemRole, String> changed = new HashMap<>();
+        Map<String, SystemRole> restricted = SystemRole.getRestricted();
+        for (RoleDto role : configuration.getRoles())
+        {
+            if (restricted.containsKey(role.getName()))
+            {
+                SystemRole systemRole = restricted.get(role.getName());
+                // get MD5 hash for the role
+                String hash = "";
+                try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
this logic better to be placed as static method in `MessageDigestUtils`

Reply:
done